server/monitor-types/snmp.js

```diff
@@ -0,0 +1,107 @@
+const { MonitorType } = require("./monitor-type");
+const { UP, DOWN, log } = require("../../src/util");
+const snmp = require("net-snmp");
+
+class SNMPMonitorType extends MonitorType {
+    name = "snmp";
+
+    /**
+     * Checks the SNMP value against the condition and control value.
+     * @param {object} monitor The monitor object associated with the check.
+     * @param {object} heartbeat The heartbeat object to update.
+     * @param {object} _server Unused server object.
+     */
+    async check(monitor, heartbeat, _server) {
+
+        log.debug("monitor", `SNMP: Community String: ${monitor.snmpCommunityString}`);
+        log.debug("monitor", `SNMP: OID: ${monitor.snmpOid}`);
+        log.debug("monitor", `SNMP: Version: ${monitor.snmpVersion}`);
+        log.debug("monitor", `SNMP: Condition: ${monitor.snmpCondition}`);
+        log.debug("monitor", `SNMP: Control Value: ${monitor.snmpControlValue}`);
+
+        const options = {
+            port: monitor.port || '161',
+            retries: monitor.maxretries,
+            timeout: 1000,
+            version: getKey(snmp.Version, monitor.snmpVersion) || snmp.Version2c,
+        };
+
+        function getKey(obj, value) {
+            return Object.keys(obj).find(key => obj[key] === value) || null;
+        }
+
+        try {
+            const session = snmp.createSession(monitor.hostname, monitor.snmpCommunityString, options);
+
+            // Handle errors during session creation
+            session.on('error', (error) => {
+                heartbeat.status = DOWN;
+                heartbeat.msg = `SNMP: Error creating SNMP session: ${error.message}`;
+                log.debug("monitor", `SNMP: ${heartbeat.msg}`);
```

Comment:
no need to have `SNMP` twice.
Also unsure if this logging is needed/benefial. Are we not doing this in the general code? 

```suggestion
                log.debug("monitor", heartbeat.msg);
```